point_cloud_test/src/lib.rs

```diff
@@ -0,0 +1,130 @@
+use point_cloud_client::PointCloudClient;
+/// This module has functions to generate synthetic point clouds in a temp dir
+/// and provides queries on these synthetic point clouds.
+use point_viewer::data_provider::{DataProviderFactory, OnDiskDataProvider};
+use point_viewer::octree::{build_octree, Octree};
+use point_viewer::read_write::{Encoding, NodeWriter, OpenMode, RawNodeWriter, S2Splitter};
+use point_viewer::s2_cells::S2Cells;
+use protobuf::Message;
+use std::fs::File;
+use std::io::BufWriter;
+use std::path::{Path, PathBuf};
+use std::sync::Once;
+use tempdir::TempDir;
+
+mod synthetic_data;
+pub use synthetic_data::{Batched, SyntheticData};
+
+mod queries;
+pub use queries::{get_abb_query, get_cell_union_query, get_frustum_query, get_obb_query};
+
+pub const S2_LEVEL: u64 = 20;
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct Arguments {
+    // Minimal precision that this point cloud should have.
+    // This decides on the number of bits used to encode each node.
+    pub resolution: f64,
+    // Width of the area to generate points for (used for local x and y), in meters.
+    pub width: f64,
+    // Height of the area to generate points for, in meters
+    pub height: f64,
+    // The number of threads used to shard octree building.
+    pub num_threads: usize,
+    // The number of points in the test point cloud.
+    pub num_points: usize,
+    // The batch size used for building the S2 point cloud.
+    pub batch_size: usize,
+    // The seed used for generating point clouds
+    pub seed: u64,
+}
+
+impl Eq for Arguments {}
+
+impl Default for Arguments {
+    fn default() -> Arguments {
+        Arguments {
+            resolution: 0.001,
+            width: 200.0,
+            height: 20.0,
+            num_threads: 10,
+            num_points: 1_000_000,
```

Comment:
as told offline, the point density might be also relevant for the query speed, maybe we can have some relevant use cases that can also be important to measure

Reply:
Not sure what you're suggesting – increasing that number, or decreasing it? I plan on making a separate binary anyway that can read those parameters from the command line, which can be used to explore what factors make querying slow or fast.